packages/devtools_app/lib/src/network/network_controller.dart

```diff
@@ -0,0 +1,204 @@
+      final json = event.toJson();
+      final id = json['id'];
+      if (id == null) {
+        continue;
+      }
+      // The start event for HTTP contains a filter key which we can use.
+      // Note: only HTTP client requests are currently logged to the timeline.
+      if ((!json['args'].containsKey('filterKey') ||
+              json['args']['filterKey'] != 'HTTP/client') &&
+          !outstandingRequestsMap.containsKey(id)) {
+        continue;
+      }
+      httpEventIds.add(id);
+    }
+
+    // Group all HTTP timeline events with the same ID.
+    final Map<String, List<Map>> httpEvents = {};
+    for (final event in events) {
+      final json = event.toJson();
+      final id = json['id'];
+      if (id == null) {
+        continue;
+      }
+      if (httpEventIds.contains(id)) {
+        if (!httpEvents.containsKey(id)) {
+          httpEvents[id] = [];
+        }
+        httpEvents[id].add(json);
+      }
+    }
+
+    // Build our list of network requests from the collected events.
+    for (final request in httpEvents.entries) {
+      final requestId = request.key;
+      final requestData = HttpRequestData.fromTimeline(
+          _timelineMicrosOffset, request.value);
+
+      // If there's a new event which matches a request that was previously in
+      // flight, update the associated HttpRequestData.
+      if (outstandingRequestsMap.containsKey(requestId)) {
+        final outstandingRequest = outstandingRequestsMap[requestId];
+        outstandingRequest.merge(requestData);
+        if (!outstandingRequest.inProgress) {
+          outstandingRequestsMap.remove(requestId);
+        }
+        continue;
+      } else if (requestData.inProgress) {
+        outstandingRequestsMap.putIfAbsent(requestId, () => requestData);
+      }
+      currentValues.add(requestData);
+    }
+    // Trigger refresh.
+    _httpRequestsNotifier.value = HttpRequests(
+        requests: currentValues, outstanding: outstandingRequestsMap);
+  }
+
+  void _startPolling() {
+    Future<void>.delayed(const Duration(milliseconds: 1000)).then((_) {
+      if (_httpRecordingNotifier.value) {
+        refreshRequests();
+        _startPolling();
+      }
+    });
+  }
+
+  Future<void> _forEachIsolate(Future Function(IsolateRef) callback) async {
+    final vm = await serviceManager.service.getVM();
+    final futures = <Future>[];
+    for (final isolate in vm.isolates) {
+      futures.add(callback(isolate));
+    }
+    await Future.wait(futures);
+  }
+
+  Future<void> _setHttpTimelineRecording(bool state) async {
+    await _forEachIsolate((isolate) async {
+      final future = serviceManager.service
+          .setHttpEnableTimelineLogging(isolate.id, state);
+      // If the isolate is paused the request above will never complete.
+      await Future.any([
+        future,
+        Future.delayed(const Duration(milliseconds: 500)),
+      ]);
+    });
+    _httpRecordingNotifier.value = state;
```

Comment:
Should this happen if some or all of the isolates fail to set up their timeline logging?

Reply:
As per my response on the other thread, this can never actually fail, only be delayed until the isolate is resumed in the future.